src/azul/terra.py

```diff
@@ -148,6 +148,13 @@ def register_with_sam(self) -> None:
             log.info('Google service account successfully registered with SAM.')
         elif response.status == 409:
             log.info('Google service account previously registered with SAM.')
+        elif response.status == 500:
+            data = response.data.decode()
+            if data and 'Cannot update googleSubjectId' in data:
+                log.warning('Unable to register %s. SAM does not allow re-registration of '
+                            'service account emails. Refer to the troubleshooting guidelines '
+                            'in the README.', self.credentials.service_account_email)
+            raise RuntimeError(response.data)
```

Comment:
Can you refactor the control flow so that it joins line 152 in case the body doesn't include the magic error message?